### ISSUE TITLE
Possibly missing ModifiedDate on QRS PUTs

### ISSUE THREAD

**Issue by hurrifan1** [qlik-oss/qlik-cli#32]
Hi all! I am trying to write a script to update some data connections in my QSEoW environment using the qlik-cli tool. I am ultimately hoping to use the `qlik qrs dataconnection update` operation. I am getting a `Status: 409 Conflict` error message.

## What I tried

This is the PowerShell script that I am using:

```powershell
qlik qrs dataconnection update fe427efc-80c7-41c3-8b3f-0d68c69b844b `
            --connectionstring 'CUSTOM CONNECT TO "provider=QvOdbcConnectorPackage.exe;UseSystemTrustStore=false;..."' `
            --name "Staffing PG database (mypc_sa_qlik)" `
            --owner-userDirectory "MyPC" `
            --owner-userId "sa_qlik" `
            --verbose
```

This is the output I get:

```
Insecure flag set, server certificate will not be verified
PUT https://mymachine:9933/jwt/qrs/dataconnection/fe427efc-80c7-41c3-8b3f-0d68c69b844b?xrfKey=3AA3DD9062E068CD
* Establishing connection to: mymachine:9933
* TLS Handshake started
* TLS Handshake done (10ms), version: TLS v1.2
* Connection established (316ms)
> Host: mymachine:9933
> User-Agent: qlik-cli/2.19.0 (windows)       
> Transfer-Encoding: chunked
> Authorization: Bearer **omitted**
> Content-Type: application/json
> Cookie: **omitted**
> Referer: https://mymachine:9933/jwt
> X-Qlik-Xrfkey: 3AA3DD9062E068CD
> Accept-Encoding: gzip
PAYLOAD:
{
  "connectionstring": "CUSTOM CONNECT TO \"provider=QvOdbcConnectorPackage.exe;UseSystemTrustStore=false;MaxVarcharSize=262144;MoneyAsDecimal=1;UseDeclareFetch=1;SSLMode=prefer;driver=postgres;maxStringLength=4096;host=xxx.xxx;useBulkReader=true;TimetzAsVarchar=1;port=5432;Fetch=200;EnableTableTypes=1;allowNonSelectQueries=false;ByteaAsLongVarBinary=0;TextAsLongVarchar=0;db=staffing;QueryTimeout=30;UseUnicode=1;FetchTSWTZasTimestamp=1;\"",
  "name": "Staffing PG database (mypc_sa_qlik)",
  "owner": {
    "userDirectory": "MyPC",
    "userId": "sa_qlik"
  }
}
< Cache-Control: private, must-revalidate, max-age=0
< Content-Type: application/json; charset=utf-8
< Date: Thu, 27 Apr 2023 01:31:51 GMT
< Expires: Thu, 27 Apr 2023 01:31:52 GMT
< Server: Microsoft-HTTPAPI/2.0
Response time: 343ms
Status: 409 Conflict
Error: [
  {
    "id": "fe427efc-80c7-41c3-8b3f-0d68c69b844b",
    "name": "Staffing PG database (mypc_sa_qlik)",
    "modifiedByUserName": "MyPC\\sa_qlik",
    "modifiedDate": "2023-04-26T21:01:47.014Z"
  }
]
```

## Looking elsewhere

I Googled around and saw that this is enough of an issue that it has its [own Qlik Support page](https://community.qlik.com/t5/Official-Support-Articles/Qlik-Sense-QRS-API-409-conflict-error-when-running-a-PUT-request/ta-p/1713614). The solution is apparently to add the `dateModified` property to the PUT body. In the qlik-cli tool, this should presumably be done using a flag for the `qlik qrs dataconnection update` operation, however there is no `dateModified` flag.

I suppose I could save the body to a JSON file, which would include the `dateModified` property, and then use the `--file` flag but it seems like I should either be able to use a date modified flag or have it happen automatically.

So am I using that operation incorrectly and misunderstanding the genesis of that error? Or does that operation need the `dateModified` property and it's just not an available flag or something?

**Comment by hurrifan1** [comment]
Also, here are my system stats:

- Qlik-cli version (using `qlik version`): **version: 2.19.0 | branch:  commit: f646537de7f85f927aae17547ee3f849837e8a6a**
- PowerShell version: **7.3.4**
- Client machine OS: **Windows 10 v21H2**
- QSEoW release: **Qlik Sense February 2023 - 14.113.2**
- QSEoW server OS: **Windows Server 2019**

**Comment by hurrifan1** [comment]
I can confirm that using the `--file` flag works:

```powershell
$modified_conn_str | ConvertTo-Json -EnumsAsStrings -Depth 30 > temp.json

qlik qrs dataconnection update $id --file temp.json

Remove-Item -Path temp.json
```

Nevertheless, it seems like this should still be possible without having to use a temp file like that.

**Comment by wadje44** [comment]
Hey @hurrifan1 

Thank you for the awesome explanation,  We will get back to you on this soon!

**Comment by wadje44** [comment]
Hi @hurrifan1 

The issue is fixed and available in latest version of qlik-cli. 
I will close the issue for now but if you still find something, feel free to reopen or file a new issue.

Thank you :)

**Comment by wennmo** [comment]
This is resolved in `v2.20.3` - https://github.com/qlik-oss/qlik-cli/releases/tag/v2.20.3

**Comment by hurrifan1** [comment]
Fantastic, thanks all! 🎉
